File: packages/graph/__fixtures__/sample-project/src/services/auth-service.ts
import type { User, AuthToken } from '../types.js';
import { hashPassword } from '../utils/hash.js';

export class AuthService {
  authenticate(user: User, password: string): AuthToken {
    const hash = hashPassword(password);
    return {
      token: hash.slice(0, 16),
      userId: user.id,
      expiresAt: new Date(Date.now() + 3600_000),
    };
  }
}

File: packages/graph/__fixtures__/multi-lang-project/src/services/auth_service.go
package services

import (
	"crypto/sha256"
	"encoding/hex"
	"fmt"
)

type AuthToken struct {
	Token string
	User  string
}

type AuthService struct {
	secret string
}

func NewAuthService(secret string) *AuthService {
	return &AuthService{secret: secret}
}

func (s *AuthService) Authenticate(username, password string) (*AuthToken, error) {
	hash := sha256.Sum256([]byte(password + s.secret))
	token := hex.EncodeToString(hash[:])
	return &AuthToken{Token: token, User: username}, nil
}

func (s *AuthService) ValidateToken(token string) bool {
	return len(token) > 0
}

var MaxSessions = 100

func hashPassword(password string) string {
	hash := sha256.Sum256([]byte(password))
	return fmt.Sprintf("%x", hash)
}
